### PR TITLE
JAMES-3693 Integration Tests for RateLimiter

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GlobalRateLimitIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+                .build())
+            .addMailet(MailetConfiguration.TO_TRANSPORT)
+            .build();
+
+        MailetContainer.Builder mailetContainer = TemporaryJamesServer.simpleMailetContainerConfiguration()
+            .putProcessor(ProcessorConfiguration.root()
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(All.class)
+                    .mailet(ToProcessor.class)
+                    .addProperty("processor", RATE_LIMIT_STATE)))
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GlobalRateLimitIntegrationTest.java:104]: Why not put GlobalRateLimit directly here?

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GlobalRateLimitIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+                .mimeMessage(message)
+                .sender(SENDER1)
+                .recipient(RECIPIENT2));
+
+        awaitAtMostOneMinute.untilAsserted(() -> jamesServer.getProbe(SpoolerProbe.class).processingFinished());
+
+        assertThat(testIMAPClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+            .login(RECIPIENT2, PASSWORD)
+            .select(TestIMAPClient.INBOX)
+            .getMessageCount(TestIMAPClient.INBOX)).isEqualTo(0);
```
Comment (chibenwa) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/GlobalRateLimitIntegrationTest.java:196]: Where is the mail? Can we have an assert on that? 

Hint: in `var/mail/error` mail repository

**Review by chibenwa — COMMENTED** [review]
Please generalise my comments accross the 3 integration test class.

Be aware I might have done related comments on https://github.com/apache/james-project/pull/853

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Rebase please :)

**Pull request merged**